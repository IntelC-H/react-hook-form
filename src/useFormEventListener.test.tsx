import useForm from './index';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';

jest.mock('./utils/onDomRemove');

describe('useFormEventListeners', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should return error type correctly and update form state', () => {
    let container = document.createElement('div');
    // @ts-ignore
    let hook;
    document.body.appendChild(container);
    const TestHook = () => {
      hook = useForm({});
      hook.watch();
      return <input name="test" ref={hook.register({ required: true })} />;
    };

    ReactDOM.render(<TestHook />, container);

    act(() => {
      const input = document.querySelector('input');
      // @ts-ignore
      input.dispatchEvent(new Event('input'));
    });

    // @ts-ignore
    expect(hook.formState).toMatchSnapshot();
  });

  it('should skip when field is not included in the validationFields', () => {
    let container = document.createElement('div');
    // @ts-ignore
    let hook;
    document.body.appendChild(container);
    const TestHook = () => {
      hook = useForm({
        validationFields: ['nothing'],
      });
      hook.watch();
      return <input name="test" ref={hook.register({ required: true })} />;
    };

    ReactDOM.render(<TestHook />, container);

    act(() => {
      const input = document.querySelector('input');
      // @ts-ignore
      input.dispatchEvent(new Event('input'));
    });

    // @ts-ignore
    expect(hook.formState).toMatchSnapshot();
  });
});
