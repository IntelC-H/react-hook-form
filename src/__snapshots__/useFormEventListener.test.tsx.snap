// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFormEventListeners should return error type correctly and update form state 1`] = `
Object {
  "dirty": true,
  "isSubmitted": false,
  "isSubmitting": false,
  "isValid": true,
  "submitCount": 0,
  "touched": Array [
    "test",
  ],
}
`;

exports[`useFormEventListeners should skip when field is not included in the validationFields 1`] = `
Object {
  "dirty": false,
  "isSubmitted": false,
  "isSubmitting": false,
  "isValid": true,
  "submitCount": 0,
  "touched": Array [],
}
`;
